src/Cookie.php

```diff
@@ -381,6 +407,27 @@ public function match(Cookie $cookie)
     }
 
     /**
+     * Validates cookie attributes.
+     *
+     * @return bool
+     */
+    public function isValid()
+    {
+        if (null === $this->valid) {
```

Comment:
Hm, not sure about this. This value should be cleared at least when the value or the max age is "modified", but even then I am not sure the performance gain worth risking inconsistency.